lib/connectors/transient_error_helper.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'utility/logger'
+require 'utility/exception_tracking'
+require 'utility/error_monitor'
+
+module Connectors
+  class TransientErrorHelper
```

Comment:
Nit: "Transient" may be a misnomer

I probably should have called this out before, but these errors may not be transient. It might be that a specific document is just "bad." That document may _always_ fail, and with the same error. It's just not indicative of a sync-wide problem. "Sufferable" or "Bearable" or "Minor" might be better terms than "Transient".

Reply:
Another possible option: Tolerable.